### ISSUE TITLE
Hang on opening files

### ISSUE THREAD

**Issue by jmezach** [microsoft/vscode#85495]
Issue Type: <b>Bug</b>

As of late I'm having issues with VS Code where it just hangs when I try to open a file in my project. This usually happens just after starting VS Code. Once I'm in this state the only thing I can do is force quit (I'm on a Mac).

I've tried disabling all my extensions, which seems to resolve the issue, but when I subsequently enable different sets of extensions I'm starting to see the issue again. I can't really pinpoint the exact extension that's causing it, which makes me think it's not an extension issue.

Sometimes after starting VS Code things do work, so it's not a consistent issue, but it happens too often now to be workable. I've delved through log files but I can't really find anything to go on.

VS Code version: Code 1.40.1 (8795a9889db74563ddd43eb0a897a2384129a619, 2019-11-13T16:47:44.719Z)
OS version: Darwin x64 19.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz (8 x 2600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.34GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
armview|ben|0.3.3
path-intellisense|chr|1.4.2
xml|Dot|2.5.0
gitlens|eam|10.2.0
nasc-touchbar|fel|1.12.0
rest-client|hum|0.23.0
ilspy-vscode|ics|0.7.10
docomment|k--|0.1.8
ecdc|mit|1.3.0
azure-pipelines|ms-|1.157.4
vscode-azurefunctions|ms-|0.20.0
vscode-docker|ms-|0.8.2
remote-containers|ms-|0.83.1
remote-ssh|ms-|0.47.2
remote-ssh-edit|ms-|0.47.2
remote-wsl|ms-|0.40.3
vscode-remote-extensionpack|ms-|0.17.0
azure-account|ms-|0.8.7
csharp|ms-|1.21.8
powershell|ms-|2019.11.0
vsliveshare|ms-|1.0.1293
vsonline|ms-|1.0.1287
team|ms-|1.161.0
azurerm-vscode-tools|msa|0.8.2
vscode-versionlens|pfl|0.24.0
vscode-yaml|red|0.6.1
vscode-coverage-gutters|rya|2.4.2
arm-snippets|sam|1.4.19
vscodeintellicode|Vis|1.2.0

</details>

**Comment by jmezach** [comment]
I've made a small recording of what I'm seeing happening:

![VS-Code-Issue](https://user-images.githubusercontent.com/1225489/69496129-0b5ec200-0ecf-11ea-895c-5c9705f63bb8.gif)

**Comment by eamodio** [comment]
@jmezach If you open up the dev tools (via _Developer: Toggle Developer Tools_ from the command palette), do you see any errors in the console when this happens?

**Comment by jmezach** [comment]
@eamodio I opened up the developer tools in one of my broken instances and all I see is the following. I don't see anything out of the ordinary. In this case I opened up a new window and then tried to open a folder, but that window immediately got stuck. Interestingly the other window from which I opened this window is still working just fine.

```
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [felipe.nasc-touchbar]: Command `workbench.action.toggleSidebarVisibility` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [felipe.nasc-touchbar]: Command `workbench.action.togglePanel` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [felipe.nasc-touchbar]: Command `workbench.action.showCommands` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [felipe.nasc-touchbar]: Command `editor.action.toggleRenderWhitespace` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [felipe.nasc-touchbar]: Command `workbench.action.openGlobalSettings` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [felipe.nasc-touchbar]: Command `git.sync` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3462 [ms-vscode.powershell]: Command `workbench.action.debug.start` appears multiple times in the `commands` section.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4286 Overwriting grammar scope name to file mapping for scope source.yaml.
Old grammar file: file:///Applications/Visual%20Studio%20Code.app/Contents/Resources/app/extensions/yaml/syntaxes/yaml.tmLanguage.json.
New grammar file: file:///Users/jmezach/.vscode/extensions/ms-azure-devops.azure-pipelines-1.157.4/syntaxes/yaml.tmLanguage.json
register @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4286
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4286 Overwriting grammar scope name to file mapping for scope source.yaml.
Old grammar file: file:///Users/jmezach/.vscode/extensions/ms-azure-devops.azure-pipelines-1.157.4/syntaxes/yaml.tmLanguage.json.
New grammar file: file:///Users/jmezach/.vscode/extensions/redhat.vscode-yaml-0.6.1/syntaxes/yaml.tmLanguage.json
register @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4286
2/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] (node:11135) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] VSCode Account: the extension commands already exist.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [vsonline][activate][info] User logged in as jonathanm@infosupport.com +950ms
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [vsonline][activate][info] Extension activating Object +2ms
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [vsonline][EnvironmentDetailsExplorer][info] Updating environment details +10ms
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [vsonline][EnvironmentDetailsExplorer][info] Updating environment details +327ms
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1541   ERR No application in the Launch Services database matches the input criteria.: Error: No application in the Launch Services database matches the input criteria.
```

**Comment by egamma** [comment]
This looks like an issue with the textmate grammar for yml files you are using that makes VS Code hang.

```
Old grammar file: file:///Users/jmezach/.vscode/extensions/ms-azure-devops.azure-pipelines-1.157.4/syntaxes/yaml.tmLanguage.json.
New grammar file: file:///Users/jmezach/.vscode/extensions/redhat.vscode-yaml-0.6.1/syntaxes/yaml.tmLanguage.json
```

Both grammars come from extensions. Can you try to narrow down which textmate grammar is causing the issue by disabling the other extension.

**Comment by jmezach** [comment]
@egamma I first disabled the RedHat YAML extension and that initially seemed to resolve the issue. Did a couple of restarts and all seemed good. Then I opened up a New Window and tried to open a project there and it hung again.

Next I disabled the Azure Pipelines extension as well, restarted VS Code and the issue was still there. Force Quitting it and then restarting it the issue seems to be gone, but now I'm still not sure which extension is causing the issue.

**Comment by egamma** [comment]
@jmezach can you share the file that causes the hang. Do you have steps for how to reproduce the hang with this file?

**Comment by jmezach** [comment]
It happens with pretty much any file I try to open, regardless of the file
type or content. And sometimes I can't even open a folder at all, it just
sits there and does nothing.

Op ma 25 nov. 2019 15:59 schreef Erich Gamma <notifications@github.com>:

**Comment by egamma** [comment]
If it happens with any file then it is unlikely a textmate grammar loop.

Can you use a different user data configuration by starting code with --user-data-dir SOME-FOLDER.

Can you reproduce this in the VS Code insider version?

Can you reproduce it when you have all extensions disabled? If not then I suggest an extension disabling bisect.

**Comment by jmezach** [comment]
I wasn't able to reproduce this when I disabled all of my extensions. Then
again when I selectively enabled them I wasn't able to pinpoint which
extension causes the issue, because I had multiple suspects. Is there some
easy way to automate this process that I'm not aware of?

I'll try with the custom user data folder tonight to see if that changes
anything. And I'll also try with Insiders to see if I can reproduce there.

Op ma 25 nov. 2019 16:37 schreef Erich Gamma <notifications@github.com>:

**Comment by jmezach** [comment]
@egamma So I just ran `code --user-data-dir /tmp/vscode-temp-profile` which seems to run with all the extensions I have installed, but without it having remembered some choices I made before. I was able to reproduce the issue that I'm unable to open a folder while running with that command.

I then installed the Insiders build and I didn't initially have the same issue, which made sense since I didn't have all of the same extensions installed. So I installed all the same extensions, restarted Code Insiders and now I'm seeing a different yet similar issue. Now when I expand a folder I get a spinner and a progress bar at the top of the explorer. Trying to open a file from an already expanded folder also results in the file not being opened and the progress bar being displayed.

So it looks like an extension is to blame, but how do we figure out which one?

**Comment by egamma** [comment]
>So it looks like an extension is to blame, but how do we figure out which one?

What is a hint that it happens when you open a file, it could be an extension that is associated with this file type. Does the hang always happen when you open a YAML file?

There is no automated process, but disabling an extension from the Extension Viewlet is straight forward.

![image](https://user-images.githubusercontent.com/172399/69638062-3fb2b980-105a-11ea-9784-16aa3aabfdf8.png)

I've looked at your extensions and the ones boxed in yellow are extensions that I have seen working (and would have caused issues if they do not work)

![image](https://user-images.githubusercontent.com/172399/69638323-c1a2e280-105a-11ea-8cff-28af86a2f790.png)
![image](https://user-images.githubusercontent.com/172399/69638416-f020bd80-105a-11ea-8a69-a6add53dd7a7.png)

So I suggest to start with the extensions that are not yellow boxed.

**Comment by jmezach** [comment]
Alright, I disabled all of the extensions you listed that weren't yellow boxed and I didn't have the issue. Then I enabled one of the other extensions, tried to open different types of files (JSON, YAML, C#, XML, etc.), restarted Visual Studio Code and opened up some more random files of different types. Then I disabled the extension that I just enabled and enabled a different one and repeated the process. I wasn't able to reproduce the issue with any of the extensions enabled, so that leads me to believe that it must be a combination of extension.

I tried enabling both the vscode-yaml and azure-pipelines extensions, since they both work with YAML files, but so far I've been unable to reproduce the issue. I'll try some more combinations here, but I have a feeling that I'm looking for a needle in a haystack.

**Comment by jmezach** [comment]
Okay I now have all of my extensions re-enabled and I'm still unable to reproduce the issue. I'm guessing the recent patch resolved the issue for me, even though I saw it happen today even after updating to the latest bits. Or at least I think I did. Anyway, it doesn't seem to happen anymore so I'm glad I can be productive again ;).

I'll close this for now. I'll re-open if I run into this issue again later on.
